app/src/main/java/com/keylesspalace/tusky/util/ImageLoadingHelper.kt

```diff
@@ -0,0 +1,43 @@
+@file:JvmName("ImageLoadingHelper")
+
+package com.keylesspalace.tusky.util
+
+import android.widget.ImageView
+import androidx.annotation.Px
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.resource.bitmap.FitCenter
+import com.bumptech.glide.load.resource.bitmap.RoundedCorners
+import com.keylesspalace.tusky.R
+
+
+private val fitCenterTransformation = FitCenter()
+
+fun loadAvatar(url: String?, imageView: ImageView, @Px radius: Int, animate: Boolean) {
+
+    if(url.isNullOrBlank()) {
+        Glide.with(imageView)
+                .load(R.drawable.avatar_default)
+                .into(imageView)
+    } else {
+        if (animate) {
+            Glide.with(imageView)
+                    .load(url)
+                    .transform(
```

Comment:
This results in two additional bitmap allocations and two expensive transformations but it seems to be the only way to do this without writing our own transformation...

Reply:
yes 😔
Its worse than our current solution, but its done on an background thread and cached, so I think we can go with it.